test/unit/org/apache/cassandra/tools/nodetool/stats/NodetoolTableStatsTest.java

```diff
@@ -237,4 +237,13 @@ public void testTopArg()
         tool.assertCleanStdErr();
         assertEquals(1, tool.getExitCode());
     }
+
+    @Test
+    public void testStatusArg()
+    {
+        ToolResult tool = ToolRunner.invokeNodetool("status", "-r");
```

Comment:
I think this bug goes a bit further than just the resolved host lookup logic. It seems like the tool is never showing anything definitive for the "Owns (effective)" column. I'll bubble this up on the ticket.

Reply:
+1 Either we resolve this here or we open a new ticket if you don't want to drag this one longer